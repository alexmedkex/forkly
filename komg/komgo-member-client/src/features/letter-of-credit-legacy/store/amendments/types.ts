import { ImmutableMap, stringOrNull } from '../../../../utils/types'
import { ILCAmendment } from '@komgo/types'
import { List, Map } from 'immutable'

export interface LetterOfCreditAmendmentStateProperties {
  byStaticId: Map<string, ILCAmendment>
  staticIds: List<string>
  error: stringOrNull
}

export type LetterOfCreditAmendmentState = ImmutableMap<LetterOfCreditAmendmentStateProperties>

export enum LetterOfCreditAmendmentActionType {
  SUBMIT_AMENDMENT_REQUEST = '@@letter-of-credit-amendments/SUBMIT_AMENDMENT_REQUEST',
  SUBMIT_AMENDMENT_SUCCESS = '@@letter-of-credit-amendments/SUBMIT_AMENDMENT_SUCCESS',
  SUBMIT_AMENDMENT_FAILURE = '@@letter-of-credit-amendments/SUBMIT_AMENDMENT_FAILURE',
  GET_AMENDMENT_REQUEST = '@@letter-of-credit-amendments/GET_AMENDMENT_REQUEST',
  GET_AMENDMENT_SUCCESS = '@@letter-of-credit-amendments/GET_AMENDMENT_SUCCESS',
  GET_AMENDMENT_FAILURE = '@@letter-of-credit-amendments/GET_AMENDMENT_FAILURE',
  ISSUE_AMENDMENT_REQUEST = '@@letter-of-credit-amendments/ISSUE_AMENDMENT_REQUEST',
  ISSUE_AMENDMENT_SUCCESS = '@@letter-of-credit-amendments/ISSUE_AMENDMENT_SUCCESS',
  ISSUE_AMENDMENT_FAILURE = '@@letter-of-credit-amendments/ISSUE_AMENDMENT_FAILURE',
  REJECT_AMENDMENT_REQUEST = '@@letter-of-credit-amendments/REJECT_AMENDMENT_REQUEST',
  REJECT_AMENDMENT_SUCCESS = '@@letter-of-credit-amendments/REJECT_AMENDMENT_SUCCESS',
  REJECT_AMENDMENT_FAILURE = '@@letter-of-credit-amendments/REJECT_AMENDMENT_FAILURE'
}
