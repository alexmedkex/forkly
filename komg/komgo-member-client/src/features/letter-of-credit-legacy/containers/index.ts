import LetterOfCreditApplication from './LetterOfCreditApplication'
import LetterOfCreditDashboard from './LetterOfCreditDashboard'
import LetterOfCreditView from './LetterOfCreditView'
import LetterOfCreditPresentation from './presentation/LetterOfCreditPresentation'
import LetterOfCreditPresentationHistory from './presentation/LetterOfCreditPresentationHistory'
import LetterOfCreditPresentationDetails from './presentation/LetterOfCreditPresentationDetails'
import LetterOfCreditPresentationFeedback from './presentation/LetterOfCreditPresentationFeedback'
import LetterOfCreditAuthorization from './LetterOfCreditAuthorization'

export {
  LetterOfCreditApplication,
  LetterOfCreditDashboard,
  LetterOfCreditView,
  LetterOfCreditPresentation,
  LetterOfCreditPresentationHistory,
  LetterOfCreditPresentationDetails,
  LetterOfCreditPresentationFeedback,
  LetterOfCreditAuthorization
}
