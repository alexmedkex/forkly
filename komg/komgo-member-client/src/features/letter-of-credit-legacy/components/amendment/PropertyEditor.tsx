import * as React from 'react'
import { Segment, Dropdown, Icon } from 'semantic-ui-react'
import { ILetterOfCredit } from '../../types/ILetterOfCredit'
import { FormikContext, connect } from 'formik'
import { fieldToLabel } from '../../constants/fieldsByStep'
import { LetterOfCreditAmendmentContext, findDiffsByTypes } from '../../containers/CreateAmendment'
import { selectInputForField } from '../../utils/selectInputForField'
import { violetBlue } from '../../../../styles/colors'
import { FIELD_ERROR_CLASSNAME } from '../../constants'
import { ILCAmendmentBase } from '@komgo/types'

interface PropertyEditorOwnProps {
  index: number
  field: keyof ILetterOfCredit | ''
  options: any[]
}

export type PropertyEditorProps = PropertyEditorOwnProps & {
  formik: FormikContext<ILCAmendmentBase>
}

export const pathToKey = (s: string): string => s.split('/')[1]
export const keyToPath = (s: string) => `/${s}`
export const selectDefaultValue = (s: string) => {
  switch (s) {
    case 'number':
      return 0
    case 'boolean':
      return false
    default:
      return ''
  }
}

// ATM is related to the LC
export const PropertyEditor: React.FC<PropertyEditorProps> = ({
  field,
  options,
  index,
  formik: { values, setFieldValue, errors }
}) => {
  const fieldDiff = values.diffs.find(v => v.path === keyToPath(field))

  const fieldHasError = Object.keys(errors).includes(field)

  const onFieldChange = (value: string) => {
    setFieldValue('diffs', values.diffs.map(v => (pathToKey(v.path) === field ? { ...v, value } : v)))
  }

  const lcDiffs = findDiffsByTypes(values.diffs, ['ILC'])
  const tradeDiffs = findDiffsByTypes(values.diffs, ['ITrade', 'ICargo'])

  return (
    <Segment>
      <div style={{ display: 'flex', justifyContent: 'space-between', paddingBottom: '10px' }}>
        <div>Field</div>
        <Icon
          name="close"
          link={true}
          style={{ color: `${violetBlue}` }}
          onClick={() => setFieldValue('diffs', [...tradeDiffs, ...lcDiffs.filter((_, idx) => index !== idx)])}
        />
      </div>
      <div style={{ maxWidth: '350px' }}>
        <LetterOfCreditAmendmentContext.Consumer>
          {(letterOfCredit: ILetterOfCredit) => (
            <Dropdown
              fluid={true}
              options={options.map(o => ({ value: o, content: fieldToLabel(o), text: fieldToLabel(o) }))}
              selection={true}
              value={field}
              className={fieldHasError ? FIELD_ERROR_CLASSNAME : ''}
              placeholder="Select an amendable field..."
              onChange={(_, { value: pathValue }: { value: string }) =>
                setFieldValue('diffs', [
                  ...tradeDiffs,
                  ...lcDiffs.map(
                    (l, idx) =>
                      index === idx
                        ? {
                            op: 'replace',
                            path: keyToPath(pathValue),
                            oldValue: letterOfCredit[pathValue],
                            value: selectDefaultValue(typeof letterOfCredit[pathValue]),
                            type: 'ILC'
                          }
                        : l
                  )
                ])
              }
            />
          )}
        </LetterOfCreditAmendmentContext.Consumer>
        {!!field && (
          <div style={{ display: 'flex', paddingTop: '15px' }}>
            <div style={{ width: '100%', paddingLeft: '10px' }}>
              Current
              {selectInputForField(field, fieldDiff)}
            </div>
            <div style={{ width: '100%', paddingLeft: '10px' }}>
              New
              {selectInputForField(field, fieldDiff, onFieldChange)}
            </div>
          </div>
        )}
      </div>
    </Segment>
  )
}

export default connect<PropertyEditorOwnProps, ILCAmendmentBase>(PropertyEditor)
