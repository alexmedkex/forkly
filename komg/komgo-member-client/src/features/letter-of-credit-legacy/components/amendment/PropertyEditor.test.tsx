import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { mount, ReactWrapper } from 'enzyme'
import PropertyEditor, { keyToPath } from './PropertyEditor'
import { Icon, Dropdown, Input } from 'semantic-ui-react'
import { fakeLetterOfCredit, fakeLetterOfCreditDiff } from '../../utils/faker'
import { LetterOfCreditAmendmentContext } from '../../containers/CreateAmendment'
import { FormikProvider } from 'formik'
import { fakeFormikContext } from '../../../../store/common/faker'
import { AVAILABLE_WITH_OPTIONS, FEES_PAYABLE_BY_OPTIONS } from '../../constants'
import { ILCAmendmentBase } from '@komgo/types'

const lc = fakeLetterOfCredit()
const setFieldValueFunc = jest.fn()

describe('PropertyEditor', () => {
  let wrapper: ReactWrapper
  const index = 0
  const feesPayableBy = 'feesPayableBy'
  const availableWith = 'availableWith'
  const invoiceRequirement = 'invoiceRequirement'
  const expiryDate = 'expiryDate'
  const expiryDateAmendment = fakeLetterOfCreditDiff({
    value: '02-01-1990',
    oldValue: lc.expiryDate as string,
    path: keyToPath(expiryDate)
  })

  describe('empty field', () => {
    const noFieldJSX = (
      <LetterOfCreditAmendmentContext.Provider value={lc}>
        <FormikProvider
          value={fakeFormikContext<ILCAmendmentBase>(
            {
              diffs: [],
              lcStaticId: lc._id,
              lcReference: lc.reference,
              version: 1
            },
            { setFieldValue: setFieldValueFunc }
          )}
        >
          <PropertyEditor index={index} options={[feesPayableBy]} field={''} />
        </FormikProvider>
      </LetterOfCreditAmendmentContext.Provider>
    )
    describe('Remove button', () => {
      beforeEach(() => {
        jest.resetAllMocks()

        wrapper = mount(noFieldJSX)
      })
      it('exists', () => {
        expect(wrapper.find(Icon).find({ name: 'close' }).length).toEqual(1)
      })
      it('calls setFieldValue when clicked', () => {
        expect(setFieldValueFunc).not.toHaveBeenCalled()

        wrapper
          .find(Icon)
          .find({ name: 'close' })
          .simulate('click')

        expect(setFieldValueFunc).toHaveBeenCalledWith('diffs', [])
      })
      it('does not affect diffs which are not type ILC', () => {
        wrapper = mount(
          <LetterOfCreditAmendmentContext.Provider value={lc}>
            <FormikProvider
              value={fakeFormikContext<ILCAmendmentBase>(
                {
                  diffs: [{ path: '/test', type: 'ITrade', value: '1', oldValue: '0', op: 'replace' }],
                  lcStaticId: lc._id,
                  lcReference: lc.reference,
                  version: 1
                },
                { setFieldValue: setFieldValueFunc }
              )}
            >
              <PropertyEditor index={index} options={[feesPayableBy]} field={''} />
            </FormikProvider>
          </LetterOfCreditAmendmentContext.Provider>
        )

        expect(setFieldValueFunc).not.toHaveBeenCalled()

        wrapper
          .find(Icon)
          .find({ name: 'close' })
          .simulate('click')

        expect(setFieldValueFunc).toHaveBeenCalledWith('diffs', [
          { oldValue: '0', op: 'replace', path: '/test', type: 'ITrade', value: '1' }
        ])
      })
    })
    it('does not show input boxes', () => {
      expect(wrapper.find(Input).length).toEqual(0)
    })
    it('matches snapshot', () => {
      expect(renderer.create(noFieldJSX).toJSON()).toMatchSnapshot()
    })
  })
  describe('dropdown', () => {
    let dropdownJsx
    beforeEach(() => {
      jest.resetAllMocks()

      dropdownJsx = (
        <LetterOfCreditAmendmentContext.Provider value={lc}>
          <FormikProvider
            value={fakeFormikContext<ILCAmendmentBase>(
              {
                lcStaticId: lc._id,
                lcReference: lc.reference,
                version: 1,
                diffs: [
                  fakeLetterOfCreditDiff({ path: keyToPath(feesPayableBy) }),
                  fakeLetterOfCreditDiff({ type: 'ITrade', path: '/buyerEtrmId' })
                ]
              },
              { setFieldValue: setFieldValueFunc }
            )}
          >
            <PropertyEditor options={[feesPayableBy]} index={index} field={''} />
          </FormikProvider>
        </LetterOfCreditAmendmentContext.Provider>
      )

      wrapper = mount(dropdownJsx)
    })
    it('calls setFieldValue when dropdown value is changed', () => {
      expect(setFieldValueFunc).not.toHaveBeenCalled()

      wrapper.find(Dropdown.Item).simulate('click')

      expect(setFieldValueFunc).toHaveBeenCalledWith('diffs', [
        {
          oldValue: 'OTHER',
          op: 'replace',
          path: '/buyerEtrmId',
          type: 'ITrade',
          value: 'SPLIT'
        },
        {
          oldValue: FEES_PAYABLE_BY_OPTIONS.APPLICANT,
          op: 'replace',
          path: keyToPath(feesPayableBy),
          value: '',
          type: 'ILC'
        }
      ])
    })
    it('matches snapshot', () => {
      expect(renderer.create(dropdownJsx).toJSON()).toMatchSnapshot()
    })
  })
  describe('field specified', () => {
    const fieldSpecifiedJSX = (
      <LetterOfCreditAmendmentContext.Provider value={lc}>
        <FormikProvider
          value={fakeFormikContext<ILCAmendmentBase>(
            {
              lcStaticId: lc._id,
              lcReference: lc.reference,
              version: 1,
              diffs: [expiryDateAmendment]
            },
            { setFieldValue: setFieldValueFunc }
          )}
        >
          <PropertyEditor options={[feesPayableBy, availableWith, expiryDate]} field={expiryDate} index={index} />
        </FormikProvider>
      </LetterOfCreditAmendmentContext.Provider>
    )
    beforeEach(() => {
      jest.resetAllMocks()

      wrapper = mount(fieldSpecifiedJSX)
    })
    afterEach(() => {
      wrapper.unmount()
    })
    it('displays 2 input boxes', () => {
      expect(wrapper.find(Input).length).toEqual(2)
    })
    it('displays old value in disabled input box', () => {
      const disabledInput = wrapper.find('input').find({ disabled: true })
      expect(disabledInput.length).toEqual(1)
      expect(disabledInput.prop('value')).toEqual(expiryDateAmendment.oldValue)
    })
    it('displays new value in other not disabled input box', () => {
      const input = wrapper.find('input').find({ value: expiryDateAmendment.value })
      expect(input.length).toEqual(1)
      expect(input.prop('disabled')).toEqual(false)
    })
    it('calls setFieldValue with correct arguments when input box is changed', () => {
      expect(setFieldValueFunc).not.toHaveBeenCalled()
      const input = wrapper.find('input').find({ value: expiryDateAmendment.value })
      input.simulate('change', { target: { value: '2019-03-02' } })
      expect(setFieldValueFunc).toHaveBeenLastCalledWith('diffs', [
        { oldValue: lc.expiryDate, op: 'replace', path: keyToPath(expiryDate), value: '2019-03-02', type: 'ILC' }
      ])
    })
    it('chooses dropdown value as field value', () => {
      expect(wrapper.find(Dropdown).prop('value')).toEqual(expiryDate)
    })
    it('displays all options in dropdown', () => {
      expect(wrapper.find(Dropdown.Item).length).toEqual(3)
    })
    it('calls setFieldValue with correct arguments when other dropdown option is changed', () => {
      expect(setFieldValueFunc).not.toHaveBeenCalled()

      const otherDropdown = wrapper.find(Dropdown.Item).find({ value: availableWith })

      otherDropdown.simulate('click')

      expect(setFieldValueFunc).toHaveBeenCalledWith('diffs', [
        { oldValue: lc[availableWith], op: 'replace', path: keyToPath(availableWith), value: '', type: 'ILC' }
      ])
    })
    it('calls setFieldValue with empty list when close icon clicked', () => {
      expect(setFieldValueFunc).not.toHaveBeenCalled()

      wrapper
        .find(Icon)
        .find({ name: 'close' })
        .simulate('click')

      expect(setFieldValueFunc).toHaveBeenCalledWith('diffs', [])
    })
    it('matches snapshot', () => {
      expect(renderer.create(fieldSpecifiedJSX).toJSON()).toMatchSnapshot()
    })
    it('makes current field red if there is an error', () => {
      const errors = { expiryDate: 'expiryDate error' }
      expect(
        renderer
          .create(
            <LetterOfCreditAmendmentContext.Provider value={lc}>
              <FormikProvider
                value={fakeFormikContext<ILCAmendmentBase>(
                  {
                    lcStaticId: lc._id,
                    lcReference: lc.reference,
                    version: 1,
                    diffs: [expiryDateAmendment]
                  },
                  { setFieldValue: setFieldValueFunc, errors }
                )}
              >
                <PropertyEditor options={[feesPayableBy, availableWith, expiryDate]} field={expiryDate} index={index} />
              </FormikProvider>
            </LetterOfCreditAmendmentContext.Provider>
          )
          .toJSON()
      ).toMatchSnapshot()
    })
  })
  describe('more than one item specified in letterOfCreditAmendments diff', () => {
    const amendments = [
      fakeLetterOfCreditDiff(),
      fakeLetterOfCreditDiff({
        value: AVAILABLE_WITH_OPTIONS.ADVISING_BANK,
        oldValue: AVAILABLE_WITH_OPTIONS.ISSUING_BANK,
        path: keyToPath(availableWith)
      }),
      fakeLetterOfCreditDiff({ value: 'ok', oldValue: 'not ok', path: keyToPath(invoiceRequirement) })
    ]

    const manyFieldsJsx = (
      <LetterOfCreditAmendmentContext.Provider value={lc}>
        <FormikProvider
          value={fakeFormikContext<ILCAmendmentBase>(
            {
              lcStaticId: lc._id,
              lcReference: lc.reference,
              version: 1,
              diffs: amendments
            },
            { setFieldValue: setFieldValueFunc }
          )}
        >
          <PropertyEditor
            options={[feesPayableBy, availableWith, invoiceRequirement]}
            field={availableWith}
            index={1}
          />
        </FormikProvider>
      </LetterOfCreditAmendmentContext.Provider>
    )
    beforeEach(() => {
      jest.resetAllMocks()

      wrapper = mount(manyFieldsJsx)
    })
    it('calls setFieldValue with the right element removed when remove icon clicked', () => {
      expect(setFieldValueFunc).not.toHaveBeenCalled()

      wrapper
        .find(Icon)
        .find({ name: 'close' })
        .simulate('click')

      expect(setFieldValueFunc).toHaveBeenCalledWith('diffs', [amendments[0], amendments[2]])
    })
    it('calls setFieldValue with the right element when input is changed', () => {
      expect(setFieldValueFunc).not.toHaveBeenCalled()

      const otherDropdownChoice = wrapper
        .find(Dropdown.Item)
        .find({ value: AVAILABLE_WITH_OPTIONS.ISSUING_BANK })
        .find({ selected: false })

      otherDropdownChoice.simulate('click')

      expect(setFieldValueFunc).toHaveBeenCalledWith('diffs', [
        amendments[0],
        { ...amendments[1], value: AVAILABLE_WITH_OPTIONS.ISSUING_BANK },
        amendments[2]
      ])
    })
    it('calls setFieldValue with the right element changed when dropdown option is changed', () => {
      expect(setFieldValueFunc).not.toHaveBeenCalled()

      wrapper
        .find(Dropdown.Item)
        .find({ value: feesPayableBy })
        .simulate('click')

      expect(setFieldValueFunc).toHaveBeenCalledWith('diffs', [
        amendments[0],
        { ...amendments[0], value: '', oldValue: lc.feesPayableBy },
        amendments[2]
      ])
    })
    it('calls setFieldValue with the right element when field on PropertyEditor is empty and dropdown item chosen', () => {
      wrapper = mount(
        <LetterOfCreditAmendmentContext.Provider value={lc}>
          <FormikProvider
            value={fakeFormikContext<ILCAmendmentBase>(
              {
                lcStaticId: lc._id,
                lcReference: lc.reference,
                version: 1,
                diffs: amendments
              },
              { setFieldValue: setFieldValueFunc }
            )}
          >
            <PropertyEditor options={[feesPayableBy, availableWith, invoiceRequirement]} index={index} field={''} />
          </FormikProvider>
        </LetterOfCreditAmendmentContext.Provider>
      )

      expect(setFieldValueFunc).not.toHaveBeenCalled()

      wrapper
        .find(Dropdown.Item)
        .find({ value: feesPayableBy })
        .simulate('click')

      expect(setFieldValueFunc).toHaveBeenCalledWith('diffs', [
        {
          ...amendments[0],
          value: '',
          oldValue: lc.feesPayableBy
        },
        amendments[1],
        amendments[2]
      ])
    })
    it('matches snapshot', () => {
      expect(renderer.create(manyFieldsJsx).toJSON()).toMatchSnapshot()
    })
  })
})
