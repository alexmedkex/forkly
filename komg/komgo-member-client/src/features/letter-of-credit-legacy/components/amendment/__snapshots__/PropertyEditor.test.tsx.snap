// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyEditor dropdown matches snapshot 1`] = `
<div
  className="ui segment"
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "paddingBottom": "10px",
      }
    }
  >
    <div>
      Field
    </div>
    <i
      aria-hidden="true"
      className="close link icon"
      onClick={[Function]}
      style={
        Object {
          "color": "#5700b5",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "maxWidth": "350px",
      }
    }
  >
    <div
      aria-expanded={false}
      className="ui fluid selection dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="default text"
        role="alert"
      >
        Select an amendable field...
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={false}
          aria-selected={true}
          className="selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Fees payable by
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PropertyEditor empty field matches snapshot 1`] = `
<div
  className="ui segment"
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "paddingBottom": "10px",
      }
    }
  >
    <div>
      Field
    </div>
    <i
      aria-hidden="true"
      className="close link icon"
      onClick={[Function]}
      style={
        Object {
          "color": "#5700b5",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "maxWidth": "350px",
      }
    }
  >
    <div
      aria-expanded={false}
      className="ui fluid selection dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="default text"
        role="alert"
      >
        Select an amendable field...
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={false}
          aria-selected={true}
          className="selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Fees payable by
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PropertyEditor field specified makes current field red if there is an error 1`] = `
<div
  className="ui segment"
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "paddingBottom": "10px",
      }
    }
  >
    <div>
      Field
    </div>
    <i
      aria-hidden="true"
      className="close link icon"
      onClick={[Function]}
      style={
        Object {
          "color": "#5700b5",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "maxWidth": "350px",
      }
    }
  >
    <div
      aria-expanded={false}
      className="ui fluid selection dropdown field error"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="text"
        role="alert"
      >
        Date of expiry
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Fees payable by
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Available with
          </span>
        </div>
        <div
          aria-checked={true}
          aria-selected={true}
          className="active selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Date of expiry
          </span>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "paddingTop": "15px",
        }
      }
    >
      <div
        style={
          Object {
            "paddingLeft": "10px",
            "width": "100%",
          }
        }
      >
        Current
        <div
          className="ui disabled input"
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <input
            disabled={true}
            onChange={[Function]}
            tabIndex={-1}
            type="date"
            value="2020-12-31"
          />
        </div>
      </div>
      <div
        style={
          Object {
            "paddingLeft": "10px",
            "width": "100%",
          }
        }
      >
        New
        <div
          className="ui input"
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <input
            disabled={false}
            onChange={[Function]}
            type="date"
            value="02-01-1990"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PropertyEditor field specified matches snapshot 1`] = `
<div
  className="ui segment"
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "paddingBottom": "10px",
      }
    }
  >
    <div>
      Field
    </div>
    <i
      aria-hidden="true"
      className="close link icon"
      onClick={[Function]}
      style={
        Object {
          "color": "#5700b5",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "maxWidth": "350px",
      }
    }
  >
    <div
      aria-expanded={false}
      className="ui fluid selection dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="text"
        role="alert"
      >
        Date of expiry
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Fees payable by
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Available with
          </span>
        </div>
        <div
          aria-checked={true}
          aria-selected={true}
          className="active selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Date of expiry
          </span>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "paddingTop": "15px",
        }
      }
    >
      <div
        style={
          Object {
            "paddingLeft": "10px",
            "width": "100%",
          }
        }
      >
        Current
        <div
          className="ui disabled input"
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <input
            disabled={true}
            onChange={[Function]}
            tabIndex={-1}
            type="date"
            value="2020-12-31"
          />
        </div>
      </div>
      <div
        style={
          Object {
            "paddingLeft": "10px",
            "width": "100%",
          }
        }
      >
        New
        <div
          className="ui input"
          style={
            Object {
              "flexGrow": 1,
            }
          }
        >
          <input
            disabled={false}
            onChange={[Function]}
            type="date"
            value="02-01-1990"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PropertyEditor more than one item specified in letterOfCreditAmendments diff matches snapshot 1`] = `
<div
  className="ui segment"
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "paddingBottom": "10px",
      }
    }
  >
    <div>
      Field
    </div>
    <i
      aria-hidden="true"
      className="close link icon"
      onClick={[Function]}
      style={
        Object {
          "color": "#5700b5",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "maxWidth": "350px",
      }
    }
  >
    <div
      aria-expanded={false}
      className="ui fluid selection dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <div
        aria-live="polite"
        className="text"
        role="alert"
      >
        Available with
      </div>
      <i
        aria-hidden="true"
        className="dropdown icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Fees payable by
          </span>
        </div>
        <div
          aria-checked={true}
          aria-selected={true}
          className="active selected item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Available with
          </span>
        </div>
        <div
          aria-checked={false}
          aria-selected={false}
          className="item"
          onClick={[Function]}
          role="option"
          style={
            Object {
              "pointerEvents": "all",
            }
          }
        >
          <span
            className="text"
          >
            Invoice requirement
          </span>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "paddingTop": "15px",
        }
      }
    >
      <div
        style={
          Object {
            "paddingLeft": "10px",
            "width": "100%",
          }
        }
      >
        Current
        <div
          aria-disabled={true}
          aria-expanded={false}
          className="ui disabled compact selection dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          style={
            Object {
              "width": "100%",
            }
          }
          tabIndex={-1}
        >
          <div
            aria-live="polite"
            className="text"
            role="alert"
          >
            Issuing bank
          </div>
          <i
            aria-hidden="true"
            className="dropdown icon"
            onClick={[Function]}
          />
          <div
            className="menu transition"
          >
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Advising bank
              </span>
            </div>
            <div
              aria-checked={true}
              aria-selected={true}
              className="active selected item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Issuing bank
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        style={
          Object {
            "paddingLeft": "10px",
            "width": "100%",
          }
        }
      >
        New
        <div
          aria-disabled={false}
          aria-expanded={false}
          className="ui compact selection dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          style={
            Object {
              "width": "100%",
            }
          }
          tabIndex={0}
        >
          <div
            aria-live="polite"
            className="text"
            role="alert"
          >
            Advising bank
          </div>
          <i
            aria-hidden="true"
            className="dropdown icon"
            onClick={[Function]}
          />
          <div
            className="menu transition"
          >
            <div
              aria-checked={true}
              aria-selected={true}
              className="active selected item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Advising bank
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                Issuing bank
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
